streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -24,14 +24,14 @@
 import static org.apache.kafka.streams.kstream.internals.WrappingNullableUtils.prepareKeySerializer;
 import static org.apache.kafka.streams.kstream.internals.WrappingNullableUtils.prepareValueSerializer;
 
-public class SinkNode<KIn, VIn, KOut, VOut> extends ProcessorNode<KIn, VIn, KOut, VOut> {
+public class SinkNode<KIn, VIn> extends ProcessorNode<KIn, VIn, Void, Void> {
```

Comment:
Here's where we declare the sink node cannot forward and hence only needs input parameters.

Reply:
That brought up a good question, as whether we need to override `getChild` and `getChildren` in `SinkNode` to throw as it should never be called?

Reply:
We already throw an exception if you try and add a child. I think it would complicate any of our processor graph traversal algorithms if we made it illegal to even call getChildren, as they would have to type-check the nodes before traversing.